wen-restart/src/wen_restart.rs

```diff
@@ -774,6 +904,48 @@ pub(crate) fn receive_restart_heaviest_fork(
     }
 }
 
+pub(crate) fn send_and_receive_heaviest_fork(
+    new_root_slot: Slot,
+    new_root_hash: Hash,
+    config: &WenRestartConfig,
+    progress: &mut WenRestartProgress,
+    pushfn: impl FnOnce(Slot, Hash),
+) -> Result<(Slot, Hash)> {
+    if config.cluster_info.id() == config.wen_restart_coordinator {
+        pushfn(new_root_slot, new_root_hash);
+        Ok((new_root_slot, new_root_hash))
+    } else {
+        let (coordinator_slot, coordinator_hash) = receive_restart_heaviest_fork(
+            config.wen_restart_coordinator,
+            config.cluster_info.clone(),
+            config.exit.clone(),
+            progress,
+        )?;
+        match verify_coordinator_heaviest_fork(
+            new_root_slot,
+            coordinator_slot,
+            &coordinator_hash,
+            config.bank_forks.clone(),
+            config.blockstore.clone(),
+            config.exit.clone(),
+            config.wen_restart_repair_slots.clone().unwrap(),
+        ) {
+            Ok(()) => pushfn(coordinator_slot, coordinator_hash),
+            Err(e) => {
+                warn!(
+                    "Failed to verify coordinator heaviest fork: {:?}, exit in 10 seconds",
+                    e
+                );
+                pushfn(new_root_slot, new_root_hash);
+                // Wait for 10 seconds so the heaviest fork gets out.
+                sleep(Duration::from_secs(10));
```

Comment:
This seems hacky, should probably either wait the actual gossip push interval or better, directly call something like `cluster_info.flush_push_queue()` to guarantee a push

Reply:
Changed to one gossip push interval.

Reply:
why not use flush_push_queue directly?

Reply:
Chatted with Greg, flush_push_queue doesn't guarantee messages being sent out, it only moves messages from local queue to the crds table. It might be possible to expose a new interface to check things have actually be sent out, but let's do that in another PR maybe, so we can start invalidator testing for this one earlier.